test/dynamo/test_misc.py

```diff
@@ -83,6 +83,12 @@
 from torch.testing._internal.jit_utils import JitTestCase
 from torch.testing._internal.logging_utils import logs_to_string
 
+try:
+    import optree
+
+    HAS_OPTREE = True
+except:
+    HAS_OPTREE = False
```

Comment:
Use importlib.utils.find_spec. Importing optree has run us into trouble when upgrading cpython versions so we try to isolate ourselves from it